src/server/api/controllers/products.controller.js

```diff
@@ -78,19 +57,48 @@ const getProductById = async (id) => {
         'is_on_discount',
         'discount_percent',
         'picture',
-      )
-      .where({ id });
-    if (products.length === 0) {
-      throw new Error(`incorrect entry with the id of ${id}`, 404);
+    )
+    .where('products.is_on_discount', '=', '1');
+
+const getProductById = async(prod_id) => {
+    const id = parseInt(prod_id, 10);
+    if (isNaN(prod_id) || id < 1) {
+        throw new HttpError('Bad request, Invalid id', 400);
+    } else {
+        try {
+            const product = await knex('products')
+                .select(
+                    'id',
+                    'name',
+                    'price',
+                    'color',
+                    'size',
+                    'is_available',
+                    'stock_amount',
+                    'is_on_discount',
+                    'discount_percent',
+                    'picture',
+                )
+                .where({ id });
+            if (product.length === 0) {
+                throw new HttpError(`Product with id ${id} doesn't exist`,
+                    404);
+            } else {
```

Comment:
Bonus:
Same here, because the `if` above throws if true, you can write it as a guard instead.